src/lib.rs

```diff
@@ -333,14 +334,9 @@ impl diff::Diff for GenericVersion {
     }
 }
 
-#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, Hash)]
-pub struct GenericHash {
-    pub hash: String,
-}
-
-impl diff::Diff for GenericHash {
+impl diff::Diff for NixHash {
```

Comment:
This sounds wrong, because `NixHash` is just a fancy string type and not a keyed container type